lte/gateway/c/core/oai/test/amf/test_amf_encode_decode.cpp

```diff
@@ -1222,6 +1222,7 @@ TEST_F(
     AmfUeContextTestServiceRequestProc,
     test_amf_service_request_without_uplinkDataStatus_RainyDay) {
   ServiceRequestMsg service_request;
+  memset(&service_request, 0, sizeof(service_request));
```

Comment:
Zero initialization should be handled by the constructor of ServiceRequestMsg
```suggestion
  ServiceRequestMsg service_request = {};
```

Reply:
ServiceRequestMsg is a defined constructor but does not have the initialization of member variables, because his members have garbage data.  ServiceRequestMsg service_request = {} is calling constructor but it was not implemented .

all nas5g library classes have defined constructors but do not have the initialization of member variables. we will take up the separate activity to clean up.

Reply:
@ganeshg87 , could you please create a GitHub issue to have well defined constructors with member initialization for Nas5G and AMF?

Reply:
ticket created. 
https://app.zenhub.com/workspaces/magma--5g-sa-60871e99b9326f001696b1ee/issues/magma/magma/10733